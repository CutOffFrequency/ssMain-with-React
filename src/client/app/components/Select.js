import React from "react";
import { subSelector } from "../services";

const Select = props => {
  let { selector, prompt, value, change, options } = props;
  let getSelector = element => subSelector(selector, element);
  return (
    <div className={getSelector("div")}>
      <p className={getSelector("p")}>{prompt}</p>
      <select className={getSelector("select")} value={value} onChange={change}>
        {options}
      </select>
    </div>
  );
};

export default Select;
