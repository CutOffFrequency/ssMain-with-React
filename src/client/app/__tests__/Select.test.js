import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Select from '../Select.react';

it('should render with props passed', () => {
    const arr = [
        <option key="one" value="one">"one"</option>,
        <option key="two" value="two">"two"</option>
    ];
    const tree = renderer.create(
        <Select val="one"
         selector="acctSelect"
         prompt="select an account"
         options={arr}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
})

it('should render children: options', () => {
    const arr = [
        <option key="one" value="one">"one"</option>,
        <option key="two" value="two">"two"</option>
    ];
    const handleChange = (e) => e.target.value;
    const component = renderer.create(
        <Select val="one"
         selector="acctSelect"
         prompt="select an account"
         options={arr}
         onSelectChange={handleChange}
        />
    );
    // initial state
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // find the select
    const sel = tree.children.filter(child => child.type === 'select')[0];
    // find children: options
    const opts = sel.children.filter(child => child.type === 'option');
    // assertions
    expect(opts.length).toBe(2);
    expect(tree).toMatchSnapshot();
    // test values
    expect(opts[0].props.value).toBe("one");
    expect(opts[1].props.value).toBe("two");
})