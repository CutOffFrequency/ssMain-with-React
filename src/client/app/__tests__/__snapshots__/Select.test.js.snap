// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render children: options 1`] = `
<div
  className="acctSelect-parentDiv"
>
  <p
    className="acctSelect p"
  >
    select an account
  </p>
  <select
    className="acctSelect select"
    onChange={[Function]}
    value="one"
  >
    <option
      value="one"
    >
      "one"
    </option>
    <option
      value="two"
    >
      "two"
    </option>
  </select>
</div>
`;

exports[`should render children: options 2`] = `
<div
  className="acctSelect-parentDiv"
>
  <p
    className="acctSelect p"
  >
    select an account
  </p>
  <select
    className="acctSelect select"
    onChange={[Function]}
    value="one"
  >
    <option
      value="one"
    >
      "one"
    </option>
    <option
      value="two"
    >
      "two"
    </option>
  </select>
</div>
`;

exports[`should render with props passed 1`] = `
<div
  className="acctSelect-parentDiv"
>
  <p
    className="acctSelect p"
  >
    select an account
  </p>
  <select
    className="acctSelect select"
    onChange={[Function]}
    value="one"
  >
    <option
      value="one"
    >
      "one"
    </option>
    <option
      value="two"
    >
      "two"
    </option>
  </select>
</div>
`;
